File: 2163-kth-distinct-string-in-an-array/kth-distinct-string-in-an-array.cpp
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        unordered_map<string, int> dict;
        for (string& elem : arr){
            dict[elem]++;
        }

        int steps = k;
        for (string& elem : arr){
            if (dict[elem] == 1){
                steps--;
                if (steps == 0)
                    return elem;
            }
        }

        return "";

    }
};

File: 2163-kth-distinct-string-in-an-array/kth-distinct-string-in-an-array.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        mp = dict()
        for s in arr:
            mp[s] = mp.get(s, 0) + 1
        
        steps = k
        for s in arr:
            if mp[s] == 1:
                steps -= 1
                if steps == 0:
                    return s

        return ""
        